### PR TITLE
DXCDT-320: Fix management of action triggers

### PR DESCRIPTION
**Pull request by willvedd**
### 🔧 Changes

This PR attempts to improve the management of action with regards to triggers. Currently, there are two major issues:

- When creating actions, the interactive list of action triggers only lists `post-login`, prohibiting the selection of any other trigger
- When updating actions, the trigger in most (if not all) cases cannot be updated, with minor exceptions

In the case of action creation, the interactive prompt should display all currently available action triggers. Currently, it only displays `post-login` erroneously. This was a case of correcting some filtering logic.

In the case of the action update, while it is technically possible to update the trigger in certain cases, the majority of the time the trigger cannot be updated. AFAICT, the guidance for changing a trigger is to create a new action. For this reason, I am proposing to remove the ability to update a trigger. Otherwise, the operation will fail in the majority of cases.

### 📝 Checklist

- [ ] All new/changed/fixed functionality is covered by tests (or N/A)
- [ ] I have added documentation for all new/changed functionality (or N/A)

### CODE REVIEW

File: docs/auth0_actions_update.md [hunk 1]
```diff
@@ -7,7 +7,7 @@ Update an action.
 
 To update interactively, use `auth0 actions update` with no arguments.
 
-To update non-interactively, supply the action id, name, trigger, secrets and dependencies through the flags.
+To update non-interactively, supply the action id, name, cod, secrets and dependencies through the flags.
```
Comment (sergiught) [inline docs/auth0_actions_update.md:10]:
```suggestion
To update non-interactively, supply the action id, name, code, secrets and dependencies through the flags.
```

**Review by sergiught — CHANGES_REQUESTED** [review]

File: docs/auth0_actions_update.md [hunk 2]
```diff
@@ -19,11 +19,11 @@ auth0 actions update [flags]
 ```
   auth0 actions update <action-id> 
   auth0 actions update <action-id> --name myaction
-  auth0 actions update <action-id> --name myaction --trigger post-login
-  auth0 actions update <action-id> --name myaction --trigger post-login --code "$(cat path/to/code.js)"
-  auth0 actions update <action-id> --name myaction --trigger post-login --code "$(cat path/to/code.js)" --dependency "lodash=4.0.0"
-  auth0 actions update <action-id> --name myaction --trigger post-login --code "$(cat path/to/code.js)" --dependency "lodash=4.0.0" --secret "SECRET=value"
-  auth0 actions update <action-id> --name myaction --trigger post-login --code "$(cat path/to/code.js)" --dependency "lodash=4.0.0" --dependency "uuid=9.0.0" --secret "API_KEY=value" --secret "SECRET=value"
+  auth0 actions update <action-id> --name myaction
```
Comment (Widcket) [inline docs/auth0_actions_update.md:22]: This one is identical to the one above.

File: internal/cli/actions.go [hunk 3]
```diff
@@ -262,15 +274,15 @@ func updateActionCmd(cli *cli) *cobra.Command {
...
+			"To update non-interactively, supply the action id, name, cod, secrets and " +
 			"dependencies through the flags.",
 		Example: `  auth0 actions update <action-id> 
   auth0 actions update <action-id> --name myaction
-  auth0 actions update <action-id> --name myaction --trigger post-login
-  auth0 actions update <action-id> --name myaction --trigger post-login --code "$(cat path/to/code.js)"
-  auth0 actions update <action-id> --name myaction --trigger post-login --code "$(cat path/to/code.js)" --dependency "lodash=4.0.0"
-  auth0 actions update <action-id> --name myaction --trigger post-login --code "$(cat path/to/code.js)" --dependency "lodash=4.0.0" --secret "SECRET=value"
-  auth0 actions update <action-id> --name myaction --trigger post-login --code "$(cat path/to/code.js)" --dependency "lodash=4.0.0" --dependency "uuid=9.0.0" --secret "API_KEY=value" --secret "SECRET=value"
+  auth0 actions update <action-id> --name myaction
```
Comment (Widcket) [inline internal/cli/actions.go:281]:
```suggestion
```
This one is identical to the one above.

Comment (sergiught) [reply internal/cli/actions.go]: Fixed in [`c3e12bd` (#597)](https://github.com/auth0/auth0-cli/pull/597/commits/c3e12bd7a9c95d4dfc699c375ad5d9ddaa9a2fb4)

**Review by sergiught — APPROVED** [review]

**Pull request merged**